tqdm/std.py

```diff
@@ -976,7 +976,10 @@ def __init__(self, iterable=None, desc=None, total=None, leave=True, file=None,
 
         if total is None and iterable is not None:
             try:
-                total = len(iterable)
+                if isinstance(iterable, enumerate):
+                    total = len(list(iterable))
+                else:
+                    total = len(iterable)
```

Comment:
I am not sure this change will be approved. By calling `list` on the `enumerate` iterable you are potentially loading an entire dataset into memory.

Reply:
You could extract the wrapped iterable and work with that instead.

Reply:
Anyhow, the standard way to use `tqdm` together with `enumerate` is to wrap with `tqdm` first and `enumerate` second:

```python
enumerate(tqdm(iterable))
```